src/jit/codegencommon.cpp

```diff
@@ -7369,10 +7398,18 @@ void CodeGen::genProfilingEnterCallback(regNumber initReg, bool* pInitRegZeroed)
     // Can't have a call until we have enough padding for rejit
     genPrologPadForReJit();
 
+#if !defined(UNIX_AMD64_ABI)
     // This will emit either
     // "call ip-relative 32-bit offset" or
     // "mov rax, helper addr; call rax"
     genEmitHelperCall(CORINFO_HELP_PROF_FCN_ENTER, 0, EA_UNKNOWN);
+#else
+    // We can use any callee trash register (other than RAX, RDI, RSI) for call target.
+    // We use R11 here. This will emit either
+    // "call ip-relative 32-bit offset" or
+    // "mov r11, helper addr; call r11"
+    genEmitHelperCall(CORINFO_HELP_PROF_FCN_ENTER, 0, EA_UNKNOWN, REG_R11);
```

Comment:
@janvorli @sivarv  Thanks for your patience.  I removed the useless registers masking for RA. I've taken a look at generated code and have seen the following:

1. callee save registers r12 & r13 are pushed in the beginning of the prolog and are used as temporary registers inside the body of a method.  Below is an example of it

```
       55                   push     rbp
       4155                 push     r13
       4154                 push     r12
       4883EC60             sub      rsp, 96
       488D6C2470           lea      rbp, [rsp+70H]
       4C8BE1               mov      r12, rcx
       4C8BEF               mov      r13, rdi
       488D7D98             lea      rdi, [rbp-68H]
       B90C000000           mov      ecx, 12
       33C0                 xor      rax, rax
       F3AB                 rep stosd 
       498BCC               mov      rcx, r12
       498BFD               mov      rdi, r13
       48896590             mov      qword ptr [rbp-70H], rsp
       48897DB0             mov      gword ptr [rbp-50H], rdi
       408875EC             mov      byte  ptr [rbp-14H], sil
       488955E0             mov      gword ptr [rbp-20H], rdx
       48894DD8             mov      gword ptr [rbp-28H], rcx
```
so we can use them for intermediate stores.

2. Homing of the argument registers has different displacements from method to method. See some examples below, so for accessing to homed registers I need additional information (fixed offsets for homing or something else):

```
       4881ECC8000000       sub      rsp, 200
       488DAC24D0000000     lea      rbp, [rsp+D0H]
       4C8BEF               mov      r13, rdi
       488DBD38FFFFFF       lea      rdi, [rbp-C8H]
       B92E000000           mov      ecx, 46
       33C0                 xor      rax, rax
       F3AB                 rep stosd 
       498BFD               mov      rdi, r13
       48897DB0             mov      gword ptr [rbp-50H], rdi
       8975F4               mov      dword ptr [rbp-0CH], esi
       8955F0               mov      dword ptr [rbp-10H], edx

```
another one

```
       4883EC38             sub      rsp, 56
       488D6C2440           lea      rbp, [rsp+40H]
       4C8BEF               mov      r13, rdi
       488D7DC0             lea      rdi, [rbp-40H]
       B90D000000           mov      ecx, 13
       33C0                 xor      rax, rax
       F3AB                 rep stosd 
       498BFD               mov      rdi, r13
       48897DE0             mov      gword ptr [rbp-20H], rdi
       8975F4               mov      dword ptr [rbp-0CH], esi
```